sycl/CMakeLists.txt

```diff
@@ -244,6 +245,7 @@ set( SYCL_TOOLCHAIN_DEPLOY_COMPONENTS
      opencl-headers
      sycl-headers
      sycl
+     opencl-aot
      pi_opencl
 )
```

Comment:
I wonder if there is a better way to deploy this tool. This implementation assumes that this tool is **required** if someone wants to build a sycl-toolchain or run SYCL tests. I don't think it a valid assumption. We better keep it optional.

@alexbatashev, are you aware of LLVM infrastructure for deploying LLVM-based toolchains https://llvm.org/docs/BuildingADistribution.html? Would it to avoid custom target?

Reply:
LLVM has convenient CMake function `add_llvm_tool` (which is used in this patch, BTW). It will add your tool to the list of LLVM tools and give you ability to build and deploy it whenever `LLVM_BUILD_TOOLS` is set to `ON`. See http://llvm.org/docs/CMake.html. The drawback of this solution is that whenever you want to deploy only one tool, you have to deploy the whole LLVM stack (this is why we have a custom target that only installs what's needed for SYCL). 

Some more thoughts on this tool:
`opencl-aot` is also a subproject. And I don't think it makes much sense. Taking into account that `opencl-aot` has almost no relation to LLVM (except for using some Support features), I think it would make more sense to move this tool to `sycl/tools` and create our own `SYCL_BUILD_TOOLS` and `add_sycl_tool`. On the other hand, if other LLVM projects want to use this tool, it would be a good idea *not* to put it under `sycl`. Then we can take some inspiration from https://github.com/intel/llvm/tree/sycl/clang-tools-extra and create some `offloading-tools-extra`  subproject.

Reply:
> LLVM has convenient CMake function `add_llvm_tool` (which is used in this patch, BTW). It will add your tool to the list of LLVM tools and give you ability to build and deploy it whenever `LLVM_BUILD_TOOLS` is set to `ON`. See http://llvm.org/docs/CMake.html. The drawback of this solution is that whenever you want to deploy only one tool, you have to deploy the whole LLVM stack (this is why we have a custom target that only installs what's needed for SYCL).
> 

LLVM documentation says that you can control which targets are installed via `LLVM_DISTRIBUTION_COMPONENTS` variable if you build `install-distribution` target.
Please, read the document I referenced in my previous comment.

I'm not saying that this tool must use CMake utilities provided by the LLVM project, but making a "SYCL tool" doesn't make sense neither. It's pure OpenCL related tool, which can be used in non-SYCL context.

Reply:
@bader, removed opencl-aot from sycl-toolchain and deploy-sycl-toolchain. Now it builds separately from them by adding ```opencl-aot``` to ```-DLLVM_EXTERNAL_PROJECTS``` and to ```-DLLVM_ENABLE_PROJECTS```, and then ```ninja opencl-aot```. Is it ok to deploy the tool like this?